### ISSUE TITLE
[Bug]: Autofill doesn't work properly when we unmerge already merged area

### ISSUE THREAD

**Issue by wszymanski** [handsontable/handsontable#10456]
### Describe the bug

After performing an unmerge on some merge area there are some problems with populating data previously covered by the merge. 

### Video/Screenshots

https://github.com/handsontable/handsontable/assets/141330/bd3ced52-a26b-403d-b53b-cb81c502e472

### Handsontable version

< 13

### Your environment

doesn't matter

**Comment by AMBudnik** [comment]
Tested using this demo https://jsfiddle.net/kvf3d2xh/2/ on Chrome 115 and Firefox 116 on macOS Ventura.

Steps to replicate based on the video

1. Unmerge the `E4` cell
2. Autofill `E4` down to `E6`
3. Autofill `E4` right to `G6`
4. Autofill `F4` down to `F6` - this is where it fails

After doing all of the above steps in pre 13.1.0 demo we are able to populate data correctly. **The issue is solved**.

I also tested a similar scenario where we

1. Unmerge cell `B2`
2. Autofill `B2` right to `C2`
3. Autofill `C2` down 

as this scenario also failed in v13.0.0. Thankfully, it also works well in the newest version.
